packages/flutter/lib/src/widgets/app.dart

```diff
@@ -185,13 +189,13 @@ class WidgetsApp extends StatefulWidget {
 
 class _WidgetsAppState extends State<WidgetsApp> implements WidgetsBindingObserver {
   GlobalObjectKey<NavigatorState> _navigator;
-  LocaleQueryData _localeData;
+  Locale _locale;
```

Comment:
if _locale was `ui.window.locale ?? widget.locale` you could avoid rebuilding the app when an app with a fixed locale experienced a system-level locale change.

Not a big optimisation on the grand scale of things, i'll grant you.